python-frontend/src/main/java/org/sonar/python/tree/PythonTreeMaker.java

```diff
@@ -871,6 +874,15 @@ public Expression expression(AstNode astNode) {
     throw new IllegalStateException("Expression " + astNode.getType() + " not correctly translated to strongly typed AST");
   }
 
+  private Expression assignmentExpression(AstNode astNode) {
+    AstNode nameNode = astNode.getFirstChild(PythonGrammar.TEST);
+    Name name = name(nameNode.getFirstDescendant(PythonGrammar.NAME));
```

Comment:
It's very suspicious to use `getFirstDescendant` in `PythonTreeMaker`.
My understanding is that the official grammar allows `namedexpr_test: test [':=' test]` but that's not consistent with [PEP-572](https://www.python.org/dev/peps/pep-0572/) which seems to indicate that only a name is allowed on the left hand side.
I would throw a `RecognitionException` if `!expression(nameNode).is(Kind.NAME)`.